onnxscript/testing/benchmark/benchmark_helpers.py

```diff
@@ -0,0 +1,682 @@
+        p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        res = p.communicate()
+        out, err = res
+        sout = out.decode("utf-8", errors="ignore")
+        serr = err.decode("utf-8", errors="ignore")
+
+        if "ONNXRuntimeError" in serr or "ONNXRuntimeError" in sout:
+            if stop_if_exception:
+                raise RuntimeError(
+                    f"Unable to continue with config {config} due to the "
+                    f"following error\n{serr}"
+                    f"\n----OUTPUT--\n{sout}"
+                )
+
+        metrics = _extract_metrics(sout)
+        if len(metrics) == 0:
+            if stop_if_exception:
+                raise BenchmarkError(
+                    f"Unable (2) to continue with config {config}, no metric was "
+                    f"collected.\n--ERROR--\n{serr}\n--OUTPUT--\n{sout}"
+                )
+            else:
+                metrics = {}
+        metrics.update(config)
+        metrics["ERROR"] = serr
+        metrics["OUTPUT"] = sout
+        metrics["CMD"] = f"[{' '.join(cmd)}]"
+        data.append(metrics)
+        if verbose > 5:
+            print("--------------- ERROR")
+            print(serr)
+        if verbose >= 10:
+            print("--------------- OUTPUT")
+            print(sout)
+
+    return data
+
+
+def common_export(
+    model: Any,
+    inputs: Sequence[Any],
+    exporter: str = "dynamo",
+    target_opset: int = 18,
+    folder: str = "",
+    filename: str = "model.onnx",
+    dynamic_shapes: Any | None = None,
+    verbose: int = 0,
+    optimization: str | None = None,
+    stats: dict[str, Any] | None = None,
+):
+    """
+    Exports a model into a folder.
+
+    Args:
+        model: model
+        exporter: script, dynamo
+        folder: folder to export into
+        filename: onnx filename
+        inputs: inputs
+        dynamic_shapes: dynamic shapes
+        target_opset: target opset
+        optimization: optimization scenario
+        verbose: verbosity
+        stats: if not None, populates this
+            dictionary with statistics about time
+
+    Returns:
+        onnx proto
+
+    """
+    import torch.onnx
+
+    if folder:
+        if not os.path.exists(folder):
+            os.mkdir(folder)
+        filename = os.path.join(folder, filename)
+
+    if verbose:
+        print(f"[common_export] start exporting with {exporter!r} in {filename!r}")
+    begin = time.perf_counter()
+    if exporter == "script":
+        torch.onnx.export(
+            model,
+            inputs,
+            filename,
+            do_constant_folding=False,
+            input_names=[f"input{i}" for i in range(len(inputs))],
+            opset_version=target_opset,
+        )
+    elif exporter == "dynamo":
+        with torch.no_grad():
+            prog = torch.onnx.dynamo_export(model, *inputs)
+        onx = prog.model_proto
+        with open(filename, "wb") as f:
+            f.write(onx.SerializeToString())
```

Comment:
Leverage `onnx.save`? This way if we want to save it to textproto or json it would be possible.

Reply:
ok, maybe we should add it to onnx documentation.